src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -2777,42 +2768,46 @@ int HwEmShim::xclCopyBO(unsigned int dst_boHandle, unsigned int src_boHandle, si
     return -1;
   }
    
-  // Disabling the m2m for timebeing as it is not working as expected. So still data is getting transferred thru DMA.
-  // Will enable this logic unless we have a clarity from the m2m hw_emu kernel. Please do not remove this code 
-  /*if ( deviceQuery(key_type::m2m) && getM2MAddress() != 0 ) {
+  // Enabling the copy buffer thru the M2M under the env control. Once the issues regarding this
+  // is resolved from the sim_m2m kernel, we will remove this env control.
+  if (getenv("ENABLE_M2M_KERNEL")) {
```

Comment:
Why don't you use one more xrt.ini variable to decide. Its better to stop using the environment variables wherever possible.

Reply:
This is very temporary and will stay for a week or so, hence went with this short cut. INI option is generally used for enabling or disabling the feature and that to for a long time. This is added for convivence for time being to facilitate the Hemants team to test and fix the issue from model side.

Reply:
as suggested, replaced the ENV with the ini param

Reply:
Thanks @venkatp-xilinx